modules/apps/sync/sync-engine/src/main/java/com/liferay/sync/engine/file/system/SyncWatchEventProcessor.java

```diff
@@ -773,18 +810,15 @@ protected void queueSyncWatchEvent(
 	}
 
 	protected void renameFile(SyncWatchEvent syncWatchEvent) throws Exception {
-		Path sourceFilePath = Paths.get(
-			syncWatchEvent.getPreviousFilePathName());
-
-		SyncFile syncFile = SyncFileService.fetchSyncFile(
-			sourceFilePath.toString());
-
 		Path targetFilePath = Paths.get(syncWatchEvent.getFilePathName());
 
 		if (sanitizeFileName(targetFilePath)) {
 			return;
 		}
 
+		SyncFile syncFile = SyncFileService.fetchSyncFile(
```

Comment:
This is what was causing those random DuplicateFolder/FileName exceptions when quickly creating folders/files. If you create "New Folder", rename it to "New FolderA", then create another "New Folder", the renamed folder gets the SyncFile db record for "New Folder" instead of "New FolderA". By querying the file key instead, we always get the correct SyncFile db record.